packages/augur-sdk/src/state/getter/Markets.ts

```diff
@@ -749,47 +765,96 @@ export class Markets {
+    const primaryCategories = params.categories.map((category) => category.toLowerCase());
 
     const allMarkets = await db.findMarkets({
       selector: {
         universe,
+        // exclude markets that are finalized or awaiting finalization
+        reportingState: {
+          $nin: [
+              MarketReportingState.AwaitingFinalization,
+              MarketReportingState.Finalized
+          ],
+        },
       },
+      fields: [ 'extraInfo', 'volume', 'marketOI' ],
     });
 
     const markets = allMarkets.map((market) => {
       const { extraInfo: extraInfoBlob } = market;
       const extraInfo = parseExtraInfo(extraInfoBlob);
 
+      let categories = extraInfo && Array.isArray(extraInfo.categories) ? extraInfo.categories : [];
+      // case-insensitive
+      categories = categories.map((category) => category.toLowerCase());
+
       return {
-        categories: extraInfo && Array.isArray(extraInfo.categories) ? extraInfo.categories : [],
+        categories,
         volume: market.volume,
+        marketOI: market.marketOI,
       };
     });
 
-    const categoryStats = categories.reduce((stats, category) => {
-      stats[category] = {
-        category,
-        numberOfMarkets: 0,
-        volume: '0',
-      };
+    const makeDefaultCategoryStats = (category: string): CategoryStat => ({
+      category,
+      numberOfMarkets: 0,
+      volume: '0',
+      openInterest: '0',
+      categories: {}, // sub-categories
+    });
+
+    const categoryStats = primaryCategories.reduce((stats, category) => {
+      stats[category] = makeDefaultCategoryStats(category);
       return stats;
     }, {} as CategoryStats);
 
     markets.forEach((market) => {
-      categories.forEach((category) => {
-        if (market.categories.indexOf(category) !== -1) {
-          const stats = categoryStats[category];
+      primaryCategories.forEach((primaryCategory) => {
+        if (market.categories.indexOf(primaryCategory) === 0) { // index 0 -> primary category
+          const stats = categoryStats[primaryCategory];
           stats.numberOfMarkets++;
           stats.volume = new BigNumber(stats.volume).plus(market.volume).toString();
+          stats.openInterest = new BigNumber(stats.openInterest).plus(market.marketOI).toString();
+
+          const secondaryCategory = market.categories[1]; // index 1 -> secondary category
+          if (secondaryCategory) {
+            let secondaryStats = stats.categories[secondaryCategory];
+            if (!secondaryStats) {
+              stats.categories[secondaryCategory] = makeDefaultCategoryStats(secondaryCategory);
+              secondaryStats = stats.categories[secondaryCategory];
+            }
+            secondaryStats.numberOfMarkets++;
+            secondaryStats.volume = new BigNumber(secondaryStats.volume).plus(market.volume).toString();
+            secondaryStats.openInterest = new BigNumber(secondaryStats.openInterest).plus(market.marketOI).toString();
+          }
```

Comment:
am I missing the place where primary's secondary stats (vol and oi) get added to the primary (vol and oi)?